crates/testing/tests/tests_2/catchup.rs

```diff
@@ -286,3 +286,63 @@ async fn test_catchup_reload() {
         .run_test::<SimpleBuilderImplementation>()
         .await;
 }
+
+#[cfg(test)]
+#[cfg_attr(async_executor_impl = "tokio", tokio::test(flavor = "multi_thread"))]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+async fn test_all_restart() {
+    use std::time::Duration;
+
+    use hotshot_example_types::node_types::{CombinedImpl, TestTypes};
+    use hotshot_testing::{
+        block_builder::SimpleBuilderImplementation,
+        completion_task::{CompletionTaskDescription, TimeBasedCompletionTaskDescription},
+        overall_safety_task::OverallSafetyPropertiesDescription,
+        spinning_task::{ChangeNode, SpinningTaskDescription, UpDown},
+        test_builder::{TestDescription, TimingData},
+    };
+
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+    let timing_data = TimingData {
+        next_view_timeout: 2000,
+        ..Default::default()
+    };
+    let mut metadata: TestDescription<TestTypes, CombinedImpl> = TestDescription::default();
+    let mut catchup_nodes = vec![];
+    for i in 1..20 {
+        catchup_nodes.push(ChangeNode {
+        idx: i,
+        updown: UpDown::Restart,
+    })};
+
+    metadata.timing_data = timing_data;
+    metadata.start_nodes = 20;
+    metadata.num_nodes_with_stake = 20;
+
+    metadata.spinning_properties = SpinningTaskDescription {
+        // Restart all the nodes in view 13
+        node_changes: vec![(13, catchup_nodes)],
+    };
+    metadata.view_sync_properties =
+        hotshot_testing::view_sync_task::ViewSyncTaskDescription::Threshold(0, 21);
```

Comment:
Any chance we could get a comment? At face value `Threshold(0, 21)` doesn't quite clarify what it's doing.

Reply:
Oh I think it should just be 20, tbh I wasn't 100% sure what it's doing but checked and it meas an amount of nodes form 0-20 can enter view sycn.  I can make a comment